zephyr/src/supp_main.c

```diff
@@ -157,11 +252,12 @@ static void start_wpa_supplicant(void)
 			exitcode = -1;
 			break;
 		}
-		wpa_s_0 = wpa_s;
-		wpa_s_0->conf->filter_ssids = 1;
-		wpa_s_0->conf->ap_scan= 1;
+		wpa_s->conf->filter_ssids = 1;
+		wpa_s->conf->ap_scan= 1;
 	}
 
+	register_dummy_socket();
```

Comment:
I would like to better understand what proble the dummy socket addition is attempting to solve.

Reply:
It's a bit complicated but let me take stab:

* Supplicant uses ELOOP mechanism to perform most tasks, and the default task is a 10s cleanup task
* So, if nothing else is active, `select/poll` is blocked for 10secs (the earliest time or one of the registered FDs is ready), in Linux this addressed as it uses sockets for communication (see below) and that will set the FD ready, so, no need to wait for 10s expiry.
    - netlink sockets for all kernel communication so whenever an event is received
    - uses sockets from user interface to core supplicant code

In Zephyr for above cases we use direct function calls, so, there is no event to unblock `select/poll`, so, for every task there is a delay, this causes the overall connection time to ~30secs, with the dummy socket making the FD ready after all tasks, this comes down to ~5secs.

Of course, this is just a WAR, we are working on a proper design that fits this wpa_supplicant design, which should be available in near future and then this WAR can be removed.

Reply:
Have you considered using a [socketpair](https://github.com/zephyrproject-rtos/zephyr/blob/main/subsys/net/lib/sockets/socketpair.c) for this purpose, sounds like a better fit for this design than UDP socket (I am aware the documentation is not the best though, it seems that only a sample is documented). Just leaving this comment as FYI, I am aware we don't have time to change this before the release.